diff --git a/crates/bevy_asset/src/lib.rs b/crates/bevy_asset/src/lib.rs
index 81deeff145..7792d9f1ed 100644
--- a/crates/bevy_asset/src/lib.rs
+++ b/crates/bevy_asset/src/lib.rs
@@ -335,7 +335,7 @@
     fn default() -> Self {
         Self {
             mode: AssetMode::Unprocessed,
-            file_path: Self::DEFAULT_UNPROCESSED_FILE_PATH.to_string(),
+            file_path: "/assets/examples".to_string(),
             processed_file_path: Self::DEFAULT_PROCESSED_FILE_PATH.to_string(),
             watch_for_changes_override: None,
             use_asset_processor_override: None,
