diff --git a/crates/bevy_audio/src/audio_output.rs b/crates/bevy_audio/src/audio_output.rs
index 15236b667..57f116bde 100644
--- a/crates/bevy_audio/src/audio_output.rs
+++ b/crates/bevy_audio/src/audio_output.rs
@@ -6,28 +6,17 @@ use bevy_asset::{Asset, Assets};
 use bevy_ecs::{prelude::*, system::SystemParam};
 use bevy_math::Vec3;
 use bevy_transform::prelude::GlobalTransform;
-use rodio::{OutputStream, OutputStreamBuilder, Sink, Source, SpatialSink};
+use rodio::{OutputStream, Sink, Source, SpatialSink};
 use tracing::warn;
 
 use crate::{AudioSink, AudioSinkPlayback};
 
 /// Used internally to play audio on the current "audio device"
-#[derive(Resource)]
+#[derive(Resource, Default)]
 pub(crate) struct AudioOutput {
     stream: Option<OutputStream>,
 }
 
-impl Default for AudioOutput {
-    fn default() -> Self {
-        let stream = OutputStreamBuilder::open_default_stream()
-            .inspect_err(|_err| {
-                warn!("No audio device found.");
-            })
-            .ok();
-        Self { stream }
-    }
-}
-
 /// Marker for internal use, to despawn entities when playback finishes.
 #[derive(Component, Default)]
 pub struct PlaybackDespawnMarker;
