diff --git a/crates/bevy_winit/src/lib.rs b/crates/bevy_winit/src/lib.rs
index 46b3e3e19..81ffad2b5 100644
--- a/crates/bevy_winit/src/lib.rs
+++ b/crates/bevy_winit/src/lib.rs
@@ -432,6 +432,11 @@ pub fn winit_runner(mut app: App) {
                     };
 
                 winit_state.low_power_event = true;
+                window_events.window_resized.send(WindowResized {
+                    window: window_entity,
+                    width: window.width(),
+                    height: window.height(),
+                });
 
                 match event {
                     WindowEvent::Resized(size) => {
