diff --git a/crates/bevy_pbr/src/light.rs b/crates/bevy_pbr/src/light.rs
index 3e8c0d451..07aa7d586 100644
--- a/crates/bevy_pbr/src/light.rs
+++ b/crates/bevy_pbr/src/light.rs
@@ -694,8 +694,8 @@ impl Default for ClusterConfig {
         // 24 depth slices, square clusters with at most 4096 total clusters
         // use max light distance as clusters max `Z`-depth, first slice extends to 5.0
         Self::FixedZ {
-            total: 4096,
-            z_slices: 24,
+            total: 128,
+            z_slices: 4,
             z_config: ClusterZConfig::default(),
             dynamic_resizing: true,
         }
