diff --git a/crates/bevy_pbr/src/cluster/mod.rs b/crates/bevy_pbr/src/cluster/mod.rs
index a8c218b44..13cc9f9c6 100644
--- a/crates/bevy_pbr/src/cluster/mod.rs
+++ b/crates/bevy_pbr/src/cluster/mod.rs
@@ -239,8 +239,8 @@ impl Default for ClusterConfig {
         // 24 depth slices, square clusters with at most 4096 total clusters
         // use max light distance as clusters max `Z`-depth, first slice extends to 5.0
         Self::FixedZ {
-            total: 4096,
-            z_slices: 24,
+            total: 128,
+            z_slices: 4,
             z_config: ClusterZConfig::default(),
             dynamic_resizing: true,
         }
