diff --git a/crates/bevy_winit/src/winit_config.rs b/crates/bevy_winit/src/winit_config.rs
index b91e25d34..48f19b708 100644
--- a/crates/bevy_winit/src/winit_config.rs
+++ b/crates/bevy_winit/src/winit_config.rs
@@ -29,14 +29,7 @@ impl WinitSettings {
     /// [`Reactive`](UpdateMode::Reactive) if windows have focus,
     /// [`ReactiveLowPower`](UpdateMode::ReactiveLowPower) otherwise.
     pub fn desktop_app() -> Self {
-        WinitSettings {
-            focused_mode: UpdateMode::Reactive {
-                wait: Duration::from_secs(5),
-            },
-            unfocused_mode: UpdateMode::ReactiveLowPower {
-                wait: Duration::from_secs(60),
-            },
-        }
+        Self::default()
     }
 
     /// Returns the current [`UpdateMode`].
